lib/grape/middleware/error.rb

```diff
@@ -46,9 +47,19 @@ def call!(env)
             raise
 
           run_rescue_handler(handler, error)
+        ensure
+          execute_ensured_block! if should_ensure?
         end
       end
 
+      def should_ensure?
+        options[:ensured]
+      end
+
+      def execute_ensured_block!
+        options[:ensured_block].call
```

Comment:
coudl be block_given? instead of should_ensure check

Reply:
Thanks @undersky0 , however `block_given?` would be false as this method is not sent a block (only ensure_block is sent the block)